benches/nexmark/nexmark_dbsp_source.rs

```diff
@@ -66,12 +71,49 @@ where
     C: Data + ZSet<Key = Event, R = W>,
 {
     fn eval(&mut self) -> C {
+        // Grab a next event, either the last event from the previous call that
+        // was saved because it couldn't yet be emitted, or the next generated
+        // event.
+        let next_event = match self.next_event.clone() {
+            Some(e) => Some(e),
+            None => self.generator.next_event().unwrap(),
+        };
```

Comment:
```suggestion
        let next_event = self.next_event
            .clone()
            .unwrap_or_else(||  self.generator.next_event().unwrap());
```

Reply:
I'd actually need something like `is_some_or_else()` here, rather than `unwrap_or_else` to simplify as you've suggested here. It's a bit confusing as `self.next_event` as an `Option<NextEvent>`, whereas `self.generator.next_event()` returns a `Result<Option<NextEvent>>` (as it can error).

So I'm trying to end up with an `Option<NextEvent>` whereas `.unwrap_or_else` implies the type is just `NextEvent` while the or else part correctly returns an `Option<NextEvent>`.

Let me know if there's still a way to do something more concise here that I've missed.

Reply:
That'd be `.or_else(|| self.generator.next_event().unwrap())` then

Reply:
Nice, that works - added to the next PR. Thanks.